### PR TITLE
Change when we can ERR_FAIL_COND and ERR_FAIL_CONV_V to static_assert

### PR DESCRIPTION
**Pull request by kuruk-mm**
In #36368 reduz said we can change some on those.

I found 4, I didn't find more.

### CODE REVIEW

**Comment by bojidar-bg** [comment]
(just a random idea) Maybe this could become `ERR_STATIC_COND`, so that the string does not have to be repeated?

**Comment by kuruk-mm** [comment]
> (just a random idea) Maybe this could become `ERR_STATIC_COND`, so that the string does not have to be repeated?

Maybe, but I would let this way if someone want to put an advice

File: core/register_core_types.cpp [hunk 1]
```diff
@@ -100,7 +100,7 @@ extern void unregister_variant_methods();
 void register_core_types() {
 
 	//consistency check
-	ERR_FAIL_COND(sizeof(Callable) > 16);
+	static_assert(sizeof(Callable) <= 16, "sizeof(Callable) <= 16");
```
Comment (akien-mga) [inline core/register_core_types.cpp]: We now use C++17 so you can drop the hardcoded error message. (I assume C++17's `static_assert` will be able to show a meaningful error by itself.)

Comment (kuruk-mm) [reply core/register_core_types.cpp]: Done, captain :+1:

**Review by neikeq — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**